ProfileAPI/ProfileAPI/Models/Profile.cs

```diff
@@ -0,0 +1,28 @@
+﻿using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace ProfileAPI.Models
+{
+    public class Profile
+    {
+        [BsonRepresentation(BsonType.ObjectId)]
```

Comment:
You probably need a [BsonId] tag here as well

Reply:
I use upn as a primary key since I have to get data from the OfficeGraph